fullstory-segment-middleware/src/main/java/com/fullstorydev/fullstory_segment_middleware/FullStorySegmentMiddleware.java

```diff
@@ -168,30 +168,38 @@ private BasePayload getNewPayloadWithFSURL(BasePayload payload, Map<String, Obje
         return null;
     }
 
-    private Map<String, Object> getSuffixedProps (Map<?, ?> props) {
+    private Map<String, Object> getSuffixedProps (Map<String, Object> props) {
         // transform props to comply with FS custom events requirement
-        // TODO: Segment should fail with circular dependency, but we should check anyway
+        // TODO: Segment should not allow with circular dependency, but we should check anyway
         Map<String, Object> mutableProps = new HashMap<>();
-        Stack<Map<?,?>> stack = new Stack<>();
+        Stack<Map<String,Object>> stack = new Stack<>();
 
         stack.push(props);
 
         while (!stack.empty()) {
-            Map<?,?> map = stack.pop();
-            for (Map.Entry<?, ?> entry : map.entrySet()) {
+            Map<String,Object> map = stack.pop();
+            for (Map.Entry<String, Object> entry : map.entrySet()) {
                 // We should only be getting String type keys, but parse the map to make sure the keys we use are all String
-                String key = String.valueOf(entry.getKey());
+                String key = entry.getKey();
```

Comment:
I think the preceding comment is unnecessary since you no longer have to parse the string?